posts/2022-06-09-apis-for-real-time-data-sharing.md

```diff
@@ -0,0 +1,96 @@
+
+![Figure 8](https://user-images.githubusercontent.com/98492452/172912756-a43fcb9b-839d-4c59-81dc-74b858b7a186.png)
+
+
+Despite the simplicity of this example, note that the sharing and lineage tracking rules  for this data are already non-trivial:
+
+
+
+* Either company can update the customer’s basic information (name, address)
+* Only Acme can update the list of Acme products the customer uses (but CoolTech can view this information)
+* Only CoolTech can update the list of CoolTech products the customer uses (but Acme can view this information)
+* All previous versions of the data need to be retained, and all changes to the data need to be tracked, so that audits can be performed to ensure GDPR, CCPA, and other regulatory program compliance by a third party
+
+The process of using Vendia is simple: given the data model above, the system automatically generates cloud hosted APIs, streaming event solutions, and data storage subsystems for both Acme and CoolTech in just a few minutes. These systems “production grade” immediately – they come with prebuilt scaling, fault tolerance, and both on-the-fly and at-rest encryption built in … no additional coding is required to achieve any of these outcomes. Each company can now set up custom authentication and authorization for as many systems as they like, even using additional nodes if they want their data to span multiple public clouds.
+
+
+## Changes Over Time
+
+If the schema evolves, for example to include additional data about the customer’s level of spend, the system automatically adjusts the implementation in a seamless fashion, preserving the existing application integrations and data. (This is one of the great advantages of combining GraphQL APIs with a compiler-driven solution that actually models the data, versus a “dumb pipe” that has no idea if or when the content it carries will break existing clients.)
+
+If the business relationships themselves evolve, for example to include a new partner, CompanyThree, Vendia can automatically deploy and synchronize additional nodes for new parties without incurring downtime or disruption to the existing system, allowing the ecosystem to seamlessly evolve to meet business needs. Additional nodes can also be added to create multi-region backups, multi-cloud application strategies, or additional operational boundaries. Advanced Vendia features, including smart contracts and tokenization, can be employed as regulatory and assurance program requirements evolve, enabling all parties to manage their data (both shared and private) with control, even in highly regulated and secured environments.
```

Comment:
Tokenization seems forward thinking - want to include it already?

Reply:
Wagner wrote it... so I'm not gonna veto him. 😄